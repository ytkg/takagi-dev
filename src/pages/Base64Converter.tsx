import { useState } from 'react';
import SEO from '../components/SEO';
import { encode, decode } from '../utils/base64';

export default function Base64Converter() {
  const [input, setInput] = useState('');
  const [output, setOutput] = useState('');
  const [error, setError] = useState('');

  const handleEncode = () => {
    try {
      const encoded = encode(input);
      setOutput(encoded);
      setError('');
    } catch (_e) {
      setError('Error encoding string. Please check your input.');
      setOutput('');
    }
  };

  const handleDecode = () => {
    try {
      const decoded = decode(input);
      setOutput(decoded);
      setError('');
    } catch (_e) {
      setError('Invalid Base64 string. Please check your input.');
      setOutput('');
    }
  };

  const handleClear = () => {
    setInput('');
    setOutput('');
    setError('');
  };

  return (
    <div className="p-8">
      <SEO
        title="Base64 Converter | takagi.dev"
        description="Encode and decode Base64 quickly online."
        path="/tools/base64-converter"
      />
      <h1 className="text-2xl font-bold mb-4">Base64 Encoder / Decoder</h1>
      <div className="grid grid-cols-1 md:grid-cols-2 gap-4">
        <div>
          <h2 className="text-lg font-semibold mb-2">Input</h2>
          <textarea
            className="w-full h-64 p-2 border rounded-md font-mono text-sm bg-gray-50 dark:bg-gray-800 dark:text-white"
            value={input}
            onChange={(e) => setInput(e.target.value)}
            placeholder='Enter string to encode/decode...'
          />
        </div>
        <div>
          <h2 className="text-lg font-semibold mb-2">Output</h2>
          <pre className="w-full h-64 p-2 border rounded-md bg-gray-50 dark:bg-gray-800 overflow-auto">
            {error ? (
              <code className="text-red-500">{error}</code>
            ) : (
              <code className="dark:text-white">{output}</code>
            )}
          </pre>
        </div>
      </div>
      <div className="mt-4 flex space-x-2">
        <button
          onClick={handleEncode}
          className="px-4 py-2 bg-blue-500 text-white rounded-md hover:bg-blue-600 transition-colors"
        >
          Encode
        </button>
        <button
          onClick={handleDecode}
          className="px-4 py-2 bg-green-500 text-white rounded-md hover:bg-green-600 transition-colors"
        >
          Decode
        </button>
        <button
          onClick={handleClear}
          className="px-4 py-2 bg-gray-500 text-white rounded-md hover:bg-gray-600 transition-colors"
        >
          Clear
        </button>
      </div>
    </div>
  );
}
