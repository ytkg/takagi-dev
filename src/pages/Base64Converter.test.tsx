import { render, screen, fireEvent } from '@testing-library/react';
import Base64Converter from './Base64Converter';
import '@testing-library/jest-dom';

describe('Base64Converter', () => {
  test('renders the component', () => {
    render(<Base64Converter />);
    expect(screen.getByText('Base64 Encoder / Decoder')).toBeInTheDocument();
  });

  test('encodes the input string correctly', () => {
    render(<Base64Converter />);
    const inputArea = screen.getByPlaceholderText('Enter string to encode/decode...');
    fireEvent.change(inputArea, { target: { value: 'hello world' } });
    fireEvent.click(screen.getByText('Encode'));
    expect(screen.getByText('aGVsbG8gd29ybGQ=')).toBeInTheDocument();
  });

  test('decodes the input string correctly', () => {
    render(<Base64Converter />);
    const inputArea = screen.getByPlaceholderText('Enter string to encode/decode...');
    fireEvent.change(inputArea, { target: { value: 'aGVsbG8gd29ybGQ=' } });
    fireEvent.click(screen.getByText('Decode'));
    expect(screen.getByText('hello world')).toBeInTheDocument();
  });

  test('shows an error for invalid base64 string on decode', () => {
    render(<Base64Converter />);
    const inputArea = screen.getByPlaceholderText('Enter string to encode/decode...');
    fireEvent.change(inputArea, { target: { value: 'invalid-base64' } });
    fireEvent.click(screen.getByText('Decode'));
    expect(screen.getByText('Invalid Base64 string. Please check your input.')).toBeInTheDocument();
  });

  test('clears the input and output', () => {
    render(<Base64Converter />);
    const inputArea = screen.getByPlaceholderText('Enter string to encode/decode...');
    fireEvent.change(inputArea, { target: { value: 'hello' } });
    fireEvent.click(screen.getByText('Encode'));
    expect(screen.getByText('aGVsbG8=')).toBeInTheDocument();
    fireEvent.click(screen.getByText('Clear'));
    expect(screen.queryByText('aGVsbG8=')).not.toBeInTheDocument();
    expect(inputArea).toHaveValue('');
  });
});
