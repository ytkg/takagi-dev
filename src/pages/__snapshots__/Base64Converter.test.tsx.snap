// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Base64Converter > matches the snapshot 1`] = `
<div>
  <div
    class="p-8"
  >
    <h1
      class="text-2xl font-bold mb-4"
    >
      Base64 Encoder / Decoder
    </h1>
    <div
      class="grid grid-cols-1 md:grid-cols-2 gap-4"
    >
      <div>
        <h2
          class="text-lg font-semibold mb-2"
        >
          Input
        </h2>
        <textarea
          class="w-full h-64 p-2 border rounded-md font-mono text-sm bg-gray-50 dark:bg-gray-800 dark:text-white"
          placeholder="Enter string to encode/decode..."
        />
      </div>
      <div>
        <h2
          class="text-lg font-semibold mb-2"
        >
          Output
        </h2>
        <pre
          class="w-full h-64 p-2 border rounded-md bg-gray-50 dark:bg-gray-800 overflow-auto"
        >
          <code
            class="dark:text-white"
          />
        </pre>
      </div>
    </div>
    <div
      class="mt-4 flex space-x-2"
    >
      <button
        class="px-4 py-2 bg-blue-500 text-white rounded-md hover:bg-blue-600 transition-colors"
      >
        Encode
      </button>
      <button
        class="px-4 py-2 bg-green-500 text-white rounded-md hover:bg-green-600 transition-colors"
      >
        Decode
      </button>
      <button
        class="px-4 py-2 bg-gray-500 text-white rounded-md hover:bg-gray-600 transition-colors"
      >
        Clear
      </button>
    </div>
  </div>
</div>
`;
